sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceClientBuilder.java

```diff
@@ -87,12 +125,161 @@ public BlobServiceClientBuilder endpoint(String endpoint) {
+    }
+
+    /**
+     * Constructs a {@link SharedKeyCredential} used to authorize requests sent to the service. Additionally, if the
+     * connection string contains `DefaultEndpointsProtocol` and `EndpointSuffix` it will set the {@link
+     * #endpoint(String) endpoint}.
+     *
+     * @param connectionString Connection string of the storage account.
+     * @return the updated BlobServiceClientBuilder
+     * @throws IllegalArgumentException If {@code connectionString} doesn't contain `AccountName` or `AccountKey`.
+     * @throws NullPointerException If {@code connectionString} is {@code null}.
+     */
+    public BlobServiceClientBuilder connectionString(String connectionString) {
+        BuilderHelper.configureConnectionString(connectionString, (accountName) -> this.accountName = accountName,
+            this::credential, this::endpoint, logger);
+
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpClient} to use for sending a receiving requests to and from the service.
+     *
+     * @param httpClient HttpClient to use for requests.
+     * @return the updated BlobServiceClientBuilder object
+     */
+    public BlobServiceClientBuilder httpClient(HttpClient httpClient) {
+        if (this.httpClient != null && httpClient == null) {
+            logger.info("'httpClient' is being set to 'null' when it was previously configured.");
+        }
+
+        this.httpClient = httpClient;
+        return this;
+    }
+
+    /**
+     * Adds a pipeline policy to apply on each request sent.
+     *
+     * @param pipelinePolicy a pipeline policy
+     * @return the updated BlobServiceClientBuilder object
+     * @throws NullPointerException If {@code pipelinePolicy} is {@code null}.
+     */
+    public BlobServiceClientBuilder addPolicy(HttpPipelinePolicy pipelinePolicy) {
+        this.additionalPolicies.add(Objects.requireNonNull(pipelinePolicy, "'pipelinePolicy' cannot be null"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpLogOptions} for service requests.
+     *
+     * @param logOptions The logging configuration to use when sending and receiving HTTP requests/responses.
+     * @return the updated BlobServiceClientBuilder object
+     * @throws NullPointerException If {@code logOptions} is {@code null}.
+     */
+    public BlobServiceClientBuilder httpLogOptions(HttpLogOptions logOptions) {
+        this.logOptions = Objects.requireNonNull(logOptions, "'logOptions' cannot be null.");
+        return this;
     }
 
-    String endpoint() {
-        return super.endpoint;
+    /**
+     * Sets the configuration object used to retrieve environment configuration values during building of the client.
+     *
+     * @param configuration Configuration store used to retrieve environment configurations.
+     * @return the updated BlobServiceClientBuilder object
+     */
+    public BlobServiceClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
```

Comment:
There are Objects.requireNotNull checks in other parameters, but not this one?

Reply:
This one has a default if the value is `null` when being used.